### ISSUE TITLE
drain before updating

### ISSUE THREAD

**Issue by cgwalters** [openshift/machine-config-operator#1038]
Today we do:

 - apply update (config + osimage)
 - cordon
 - drain
 - reboot

But it's going to be somewhat common that a drain will fail; our test suite today bumps `maxUnavailable` for workers to 3, but that no longer works because the router uses a PDB that doesn't tolerate it.

It also leaves us in a bit of a messy situation because we are in a half-upgraded state potentially.

I think instead...we should clearly separate draining from config updating, and do it *first*.  We should probably do a drain on...hmm, half of the `maxUnavailable` at the same time?  And if draining fails on a node but succeeds on another, uncordon the failed-drain node cleanly.

Basically, in the `maxUnavailable=3` and 3 workers case, we can only drain 1 node at a time.  We would discover this "dynamically" by trying to drain 2 at once, failing on the router PDB, and falling back to draining just one.

Maybe we could also target draining pods with a PDB *first* - if those are going to be violated then there's no point in draining anything else.

**Comment by cgwalters** [comment]
This was fixed by https://github.com/openshift/machine-config-operator/pull/1200
